src/server/routes/import-entity/import-routes.js

```diff
@@ -0,0 +1,205 @@
+	const props = generateProps(req, res, {token});
+	const {DiscardImportEntityPage} = importEntityPages;
+	const markup = ReactDOMServer.renderToString(
+		<Layout {...propHelpers.extractLayoutProps(props)}>
+			<DiscardImportEntityPage
+				importEntity={props.importEntity}
+				token={props.token}
+			/>
+		</Layout>
+	);
+
+	res.render('target', {
+		markup,
+		props: escapeProps(props),
+		script: '/js/import-entity/discard-import-entity.js'
+	});
+}
+
+export function handleDiscardImportEntity(req, res) {
+	const {headers} = req;
+	const token = headers['x-auth-token'];
+
+	if (token !== req.session.token) {
+		res.status(401).send({error: 'No token provided.'});
+	}
+
+	const {orm} = req.app.locals;
+	const editorId = req.session.passport.user.id;
+	const {importEntity} = res.locals;
+	orm.bookshelf.transaction(async transacting => {
+		try {
+			await orm.func.imports.castDiscardVote(
+				transacting, importEntity.importId, editorId
+			);
+			// To-do: Add code to remove importEntity from the search index
+			res.status(200).send();
+		}
+		catch (err) {
+			res.status(400).send({error: err});
+		}
+	});
+}
+
+export async function approveImportEntity(req, res) {
+	const {orm} = res.app.locals;
+	const editorId = req.session.passport.user.id;
+	const {importEntity} = res.locals;
+	const entity = await orm.bookshelf.transaction(transacting =>
+		orm.func.imports.approveImport(
+			{editorId, importEntity, orm, transacting}
+		));
+	const entityUrl = getEntityUrl(entity);
+
+	/* Add code to remove import and add the newly created entity to the elastic
+		search index and remove delete import */
+
+	// Update editor achievement
+	entity.alert = (await achievement.processEdit(
+		orm, editorId, entity.revisionId
+	)).alert;
+
+	// Cleanup search indexing
+	search.indexEntity(entity);
+	// Todo: Add functionality to remove imports from ES index upon deletion
+
+	res.redirect(entityUrl);
+}
+
+export function editImportEntity(req, res) {
+	const {importEntity} = res.locals;
+	const initialState = entityToFormState(importEntity);
+	const importEntityProps = generateImportEntityProps(
+		req, res, initialState, {}
+	);
+	const {markup, props} = entityEditorMarkup(importEntityProps);
+	return res.render('target', {
+		markup,
+		props: escapeProps(props),
+		script: '/js/entity-editor.js',
+		title: 'Edit Work Import'
+	});
+}
+
+export async function approveImportPostEditing(req, res) {
+	const {orm} = req.app.locals;
+	const {importEntity} = res.locals;
+	const editorId = req.session.passport.user.id;
+	const {importId, type} = importEntity;
+	const formData = req.body;
+
+	const validateForm = getValidator(type.toLowerCase());
+
+	if (!validateForm(formData)) {
+		const err = new error.FormSubmissionError();
+		error.sendErrorAsJSON(res, err);
+	}
+
+	const entityData = transformForm[type](formData);
+
+	const entity = await orm.bookshelf.transaction(async (transacting) => {
+		await orm.func.imports.deleteImport(
```

Comment:
Need to swap the order of these, and pass the new entity BBID to the `deleteImport` function.

Reply:
Why do we need to pass the new entity BBID to the deleteImport function? Also how does it matter we keep deletion before creation or creation before deletion, as it all happens in the same transaction?